src/lib/Preview.js

```diff
@@ -974,6 +974,9 @@ class Preview extends EventEmitter {
         // Add the response interceptor to the preview instance
         this.options.responseInterceptor = options.responseInterceptor;
 
+        // Add the token generator to refresh the token if necessary
+        this.options.tokenGenerator = options.token;
```

Comment:
It seems like `previewOptions.token` is can already be the token or token generator, can we re-use that?

Reply:
`previewOptions.token` is the token (string) generated by token generator and it (the string) will be passed into viewer to generate urls and headers, etc. So we cannot merge token and generator into one variable.